plugin/src/test/java/com/google/errorprone/xplat/checker/UnnecessaryConcurrentHashMapTest.java

```diff
@@ -44,4 +45,129 @@ public void negativeCases() {
     compilationHelper.addSourceFile("UnnecessaryConcurrentHashMapTestNegative.java").doTest();
   }
 
+  @Test
+  public void refactorSameLine() {
+    BugCheckerRefactoringTestHelper.newInstance(new UnnecessaryConcurrentHashMap(), getClass())
+        .addInputLines("Test.java",
+            "import java.util.concurrent.ConcurrentHashMap;",
+            "class Test {",
+            "  private void test() {",
+            "    ConcurrentHashMap<String, Integer> map = new ConcurrentHashMap<String, Integer>();",
+            "  }",
+            "}")
+        .addOutputLines("Test.java",
+            "import java.util.Collections;",
+            "import java.util.HashMap;",
+            "import java.util.Map;",
+            "import java.util.concurrent.ConcurrentHashMap;",
+            "class Test {",
+            "  private void test() {",
+            "    Map<String, Integer> map = Collections.synchronizedMap(new HashMap<>());",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void refactorSameLineConcurrentMap() {
+    BugCheckerRefactoringTestHelper.newInstance(new UnnecessaryConcurrentHashMap(), getClass())
+        .addInputLines("Test.java",
+            "import java.util.concurrent.ConcurrentMap;",
+            "import java.util.concurrent.ConcurrentHashMap;",
+            "class Test {",
+            "  private void test() {",
+            "    ConcurrentMap<String, Integer> map = new ConcurrentHashMap<String, Integer>();",
+            "  }",
+            "}")
+        .addOutputLines("Test.java",
+            "import java.util.Collections;",
+            "import java.util.HashMap;",
+            "import java.util.Map;",
+            "import java.util.concurrent.ConcurrentHashMap;",
+            "import java.util.concurrent.ConcurrentMap;",
+            "class Test {",
+            "  private void test() {",
+            "    Map<String, Integer> map = Collections.synchronizedMap(new HashMap<>());",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void refactorDeclaration() {
+    BugCheckerRefactoringTestHelper.newInstance(new UnnecessaryConcurrentHashMap(), getClass())
+        .addInputLines("Test.java",
+            "import java.util.concurrent.ConcurrentHashMap;",
+            "class Test {",
+            "  private void test() {",
+            "    ConcurrentHashMap<String, Integer> map;",
+            "  }",
+            "}")
+        .addOutputLines("Test.java",
+            "import java.util.Map;",
+            "import java.util.concurrent.ConcurrentHashMap;",
+            "class Test {",
+            "  private void test() {",
+            "    Map<String, Integer> map;",
+            "  }",
+            "}")
+        .doTest();
+  }
+
```

Comment:
It's probally worth having a test for

Map<String, Integer> bob = new ConcurrentHashMap<>();

This is actually the much more common pattern (variable declared as interface type but is assigned a particular type map (concurrenthashmap in this case)

Reply:
I added a refactoring test for this case